p2p/src/peers.rs

```diff
@@ -682,9 +688,10 @@ impl NetAdapter for Peers {
 				ban_reason: ReasonForBan::None,
 				last_connected: Utc::now().timestamp(),
 			};
-			if let Err(e) = self.save_peer(&peer) {
-				error!("Could not save received peer address: {:?}", e);
-			}
+			to_save.push(peer);
+		}
+		if let Err(e) = self.save_peers(to_save) {
```

Comment:
Is there anything wrong if we end up saving no peers because this one failed?

Reply:
It's not ideal, but how would you know which one failed?